podcasts/DownloadManager.swift

```diff
@@ -258,7 +262,14 @@ class DownloadManager: NSObject, FilePathProtocol {
         }
     }
 
-    private func performAddToQueue(episode: BaseEpisode, url: String, previousDownloadFailed: Bool, fireNotification: Bool, autoDownloadStatus: AutoDownloadStatus) {
+    private func performAddToQueue(episode: BaseEpisode, url: String, previousDownloadFailed: Bool, fireNotification: Bool, autoDownloadStatus: AutoDownloadStatus) async {
+
+        if let task = await existingTask(for: episode) {
```

Comment:
A bit of an edge case but could tasks exists for the same episode, but downloading different URLs because the episode media changed? The Creator update the audio or the link was changed.

Reply:
I considered this and thought some future changes to cancel old episode downloads could handle this. What would you think about cancelling any existing tasks which have a different URL in this spot?

Reply:
If we find tasks that don't match the current episode url then we can cancel them.

Reply:
Since I changed to only check the current session from [your other comment](https://github.com/Automattic/pocket-casts-ios/pull/1780#discussion_r1613205988), I decided only to cancel tasks found in that session instead of checking all of them. I'm planning a set of changes to cancel tasks on refresh which don't match current episode URLs so that should handle the rest. This is really just a fallback.